backends/qualcomm/runtime/QnnManager.cpp

```diff
@@ -681,6 +682,36 @@ Error QnnManager::Compile(
     return Error::Internal;
   }
 
+  // Write context binary to file for debugging purpose
+  const char* et_qnn_debug_dir = getenv("ET_QNN_DEBUG_DIR");
+  if (et_qnn_debug_dir != nullptr) {
+    namespace fs = std::filesystem;
+
+    fs::path debug_dir = et_qnn_debug_dir;
+    if (!fs::exists(debug_dir) || !fs::is_directory(debug_dir)) {
+      QNN_EXECUTORCH_LOG_WARN(
+          "ET_QNN_DEBUG_DIR %s is not a valid directory", et_qnn_debug_dir);
+      return Error::Ok;
+    }
+
+    QnnExecuTorchContextBinary qnn_executorch_context_binary{nullptr, 0};
+    if (GetContextBinary(qnn_executorch_context_binary) != Error::Ok) {
+      return Error::Ok;
+    }
+
+    std::string binary_file = std::string(et_qnn_debug_dir) + "/qnn_context_binary.bin";
+    std::ofstream fout(binary_file, std::ios::binary);
+    fout.write(
+        reinterpret_cast<const char*>(qnn_executorch_context_binary.buffer),
```

Comment:
If you would like to dump qnn context binary, you should [use GetBinaryInfo to deserialize the flatbuffer](https://github.com/pytorch/executorch/blob/d679ad70a7745c60df581d6b110e6f79c389feb9/backends/qualcomm/runtime/backends/QnnContextCommon.cpp#L49) and use binary_info->data()->data().